src/upgrades/v0_9_0/ca_objects_migration.rs

```diff
@@ -30,52 +29,47 @@ use crate::{
     },
     pubd::RepositoryManager,
     upgrades::v0_9_0::{old_commands::*, old_events::*},
-    upgrades::{UpgradeError, UpgradeResult, UpgradeStore, MIGRATION_SCOPE},
+    upgrades::{UpgradeError, UpgradeMode, UpgradeResult, UpgradeStore, MIGRATION_SCOPE},
 };
 
 /// Migrate the current objects for each CA into the CaObjectStore
 pub struct CaObjectsMigration;
 
 impl CaObjectsMigration {
-    pub fn migrate(config: Arc<Config>, repo_manager: RepositoryManager) -> UpgradeResult<()> {
+    pub fn prepare(mode: UpgradeMode, config: Arc<Config>, repo_manager: RepositoryManager) -> UpgradeResult<()> {
         let repo_manager = Arc::new(repo_manager);
-        let store = KeyValueStore::disk(&config.data_dir, CASERVER_DIR)?;
-        let ca_store = AggregateStore::<ca::CertAuth>::disk(&config.data_dir, CASERVER_DIR)?;
+        let current_kv_store = KeyValueStore::disk(&config.data_dir, CASERVER_DIR)?;
+        let new_kv_store = KeyValueStore::disk(&config.upgrade_data_dir(), CASERVER_DIR)?;
+        let new_agg_store = AggregateStore::<ca::CertAuth>::disk(&config.upgrade_data_dir(), CASERVER_DIR)?;
 
         let signer = Arc::new(KrillSigner::build(&config.data_dir)?);
 
-        if store.version_is_before(KrillVersion::release(0, 6, 0))? {
-            Err(UpgradeError::custom("Cannot upgrade Krill installations from before version 0.6.0. Please upgrade to any version ranging from 0.6.0 to 0.8.1 first, and then upgrade to this version."))
-        } else if store.version_is_before(KrillVersion::candidate(0, 9, 0, 1))? {
-            info!("Krill version is older than 0.9.0-RC1, will now upgrade data structures.");
-
-            // Populate object store which will contain all objects produced by CAs, while we are
-            // at it.. return the information we will need in case we need to convert embedded child-parent
-            // CA relationships to use RFC 6492.
-            let derived_embedded_ca_info_map = Self::populate_ca_objects_store(config, repo_manager.clone(), signer)?;
-
-            // Migrate existing CAs to the new data structure, commands and events
-            CasStoreMigration {
-                store,
-                ca_store,
-                repo_manager,
-                derived_embedded_ca_info_map,
-            }
-            .migrate()
-        } else {
-            debug!("Krill is up to date, no CA data migration needed.");
-            Ok(())
+        info!("Krill version is older than 0.9.0, will now upgrade CA data structures.");
```

Comment:
In the old code this statement was checked, now it is not. We could be called without this being the case. Is this the right place for this log statement now?

Reply:
should not mention the version again - modified